Sync from JHUGenMELA

hroskes opened the pull request.

hroskes committed:
sync MELA with JHUGen

hroskes committed:
fix more

hroskes committed:
sync MELA with JHUGen

yaofuzhou committed:
"1" removed from MELA/parameters

"1" removed from "stop 1" MELA/mod_parameters

hroskes committed:
sync from JHUGen

yaofuzhou committed:
minor changes in STOP

yaofuzhou committed:
minor changes in STOP

hroskes committed:
merge

yaofuzhou committed:
minor changes

yaofuzhou committed:
1

1

yaofuzhou committed:
minor changes

yaofuzhou committed:
merge

yaofuzhou committed:
more functions in mod_Parameters

yaofuzhou committed:
rearranging

yaofuzhou committed:
rearranging

yaofuzhou committed:
moving alpha_s to from kinematics to parameters

moving alpha_s to from kinematics to parameters

yaofuzhou committed:
adding --no-check-certificate in this line after wget like the message recommends.

hroskes committed:
new collier URL

hroskes committed:
version number to v7.2.0

yaofuzhou committed:
Z > nunu fixed in gg(Process=51)

Z > nunu fixed in gg(Process=51)

hroskes committed:
merge

hroskes committed:
increment version number

hroskes committed:
increment version number

hroskes committed:
increment version number

hroskes committed:
sync MELA with JHUGen

hroskes committed:
incrementing to v7.2.4

yaofuzhou committed:
mod_Parameters.F90 sync'ed to MELA

hroskes committed:
merge

usarica reviewed (changes requested):
There does not seem to be any need to move EvalAlphaS function. I also do not see this need in the original generator. What was the reason to move?

hroskes commented:
I think you already know the answer: I moved it here because Yaofu moved in JHUGen.  For the reason for that, you'll have to ask him.  Does it really matter where it is?

usarica commented:
We should avoid making unnecessary changes, and also collect functions where they belong. Otherwise we are going to bloat mod_Parameters in the future to an incomprehensible module. Under this consideration of the future, yes, it does matter a bit.

usarica committed:
Add new hash

hroskes committed:
merge

hroskes committed:
renormalization/factorization scale scheme for pT of the leading jet

hroskes committed:
add option for pTJ to MELA

hroskes committed:
fixes

usarica committed:
Implement style comment and add more commentary

hroskes committed:
and another option for the softest jet

hroskes committed:
fix for smallest pT

usarica committed:
No need for numeric limits

hroskes committed:
reference for scale schemes

(and manually checked that the scales are right)

hroskes committed:
merge

hroskes committed:
fix logic

min(-1, p[c].Pt()) = -1

usarica committed:
Revert "reference for scale schemes"

This reverts commit 964f6c1ccbfd160fef154cdd40cea53023053072.

hroskes committed:
random deterministic order for testME_all.py

usarica committed:
setup and makefiles improved

usarica committed:
EventScaleScheme shall be int type

usarica committed:
Add checks for ren. and fac. scale, avoiding the change in many other reference files

usarica committed:
Doesn't matter really, but fix initialization off ZZME

hroskes committed:
Fix memory leak in MELA

it should be delete[] for xGrid and yGrid
but better to just remove the news and deletes entirely

Co-authored-by: meng xiao <xiaomengphy@gmail.com>

hroskes committed:
no need for xGrid and yGrid at all

hroskes committed:
reference for JVBF

hroskes committed:
reference for JVBF

usarica committed:
merge
